test/csuite/parse_opts/main.c

```diff
@@ -0,0 +1,315 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "test_util.h"
+
+/*
+ * Unit test testutil_parse_opts and testutil_parse_opt functions.
+ */
+
+extern char *__wt_optarg; /* option argument */
+extern int __wt_optind;   /* argv position, needed to reset __wt_getopt parser */
+extern int __wt_optreset; /* needed to reset the parser internal state */
+
+/*
+ * This structure aids in testing testutil_parse_opt. That function is useful for test applications
+ * that wish to extend/modify the basic option set provided by testutil.
+ */
+typedef struct {
+    char *c_option;
+    bool d_option;
+    bool e_option;
+    int f_option;
+} EXTENDED_OPTS;
+
+/*
+ * This drives the testing. For each given command_line, there is a matching expected TEST_OPTS
+ * structure. When argv[0] is "parse_opts", we are driving testutil_parse_opts. If argv[0] is
+ * "parse_opt", then we are parsing some of our "own" options, put into an EXTENDED_OPTS struct, and
+ * using testutil_parse_opt to parse any we don't recognize, those are put into TEST_OPTS.
```

Comment:
The use of `opts` and `opt` is subtle and confusing to keep separate in my mind when reading this. Is there a naming convention that could help? Something like `parse_opts` and `parse_single_opt` (and similar struct naming)? Or `parse_multiple` and `parse_opt`? I'm not sure what makes the most sense.

Reply:
The existing function is parse_opts, and I didn't want to change that (along with the couple dozen programs that already use it).  I'm okay with parse_single_opt.

Reply:
Fixed.